Sources/Benchmark/BenchmarkResult.swift

```diff
@@ -352,130 +352,145 @@ public struct BenchmarkResult: Codable, Comparable, Equatable {
-                                                                    relative: true,
-                                                                    units: scalingFactor))
-                    thresholdViolated = true
+            let absoluteDifference = (reverseComparison ? -1 : 1) * (lhs - rhs)
+            let relativeDifference = (reverseComparison ? 1 : -1) * (rhs != 0 ? (100 - (100.0 * Double(lhs) / Double(rhs))) : 0.0)
+
+            if let threshold = thresholds.relative[percentile], !(-threshold ... threshold).contains(relativeDifference) {
+                let deviation = ThresholdDeviation(name: name,
+                                                   target: target,
+                                                   metric: metric,
+                                                   percentile: percentile,
+                                                   baseValue: normalize(lhs),
+                                                   comparisonValue: normalize(rhs),
+                                                   difference: Int(Statistics.roundToDecimalplaces(abs(relativeDifference), 1)),
+                                                   differenceThreshold: Int(threshold),
+                                                   relative: true,
+                                                   units: scalingFactor)
+                if relativeDifference > threshold {
+                    thresholdResults.regressions.append(deviation)
+                } else if relativeDifference < -threshold {
+                    thresholdResults.improvements.append(deviation)
                 }
             }
 
-            if let threshold = thresholds.absolute[percentile] {
-                if reverseComparison ? -absoluteDifference > threshold : absoluteDifference > threshold {
-                    violationDescriptions.append(ThresholdDeviation(name: name,
-                                                                    target: target,
-                                                                    metric: metric,
-                                                                    percentile: percentile,
-                                                                    baseValue: normalize(lhs),
-                                                                    comparisonValue: normalize(rhs),
-                                                                    difference: normalize(absoluteDifference),
-                                                                    differenceThreshold: normalize(threshold),
-                                                                    relative: false,
-                                                                    units: scalingFactor))
-                    thresholdViolated = true
+            if let threshold = thresholds.absolute[percentile], abs(absoluteDifference) > threshold {
```

Comment:
```suggestion
            if let threshold = thresholds.absolute[percentile], !(-threshold ... threshold).contains(absoluteDifference) {
```
for consistency with above check for relative diff